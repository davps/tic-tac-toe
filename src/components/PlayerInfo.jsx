import React from 'react';
import PropTypes from 'prop-types';
import { MOVE } from '../constants';

/**
 * Component to display information about a player
 * @param {string} player The player value
 */

/* eslint-disable react/jsx-one-expression-per-line */
const PlayerInfo = ({ player }) => (
  <span>{` ${MOVE[player].name} (${MOVE[player].label}) `}</span>
);
/* eslint-enable */

PlayerInfo.propTypes = {
  player: PropTypes.oneOf([MOVE.PLAYER_1.val, MOVE.PLAYER_2.val]).isRequired
};

export default PlayerInfo;
