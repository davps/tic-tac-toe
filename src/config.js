/**
 * I used the key property on this data structure because it is
 * easiert to debug with it (instead of using null or empty)
 */
export const MOVE = {
  PENDING: { val: 'PENDING', label: ' ' },
  PLAYER_1: { val: 'PLAYER_1', label: 'X', name: 'Jacob' },
  PLAYER_2: { val: 'PLAYER_2', label: 'O', name: 'David' }
};

export default {};
