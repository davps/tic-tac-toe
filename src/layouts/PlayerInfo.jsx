import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { MOVE } from '../config';

/**
 * Component to display information about a player
 * @param {string} player The player value
 */
const PlayerInfo = ({ player }) => {
  return (
    <span>
      {' ' + MOVE[player].name} ( {MOVE[player].label} )
    </span>
  );
};

PlayerInfo.propTypes = {
  player: PropTypes.oneOf([MOVE.PLAYER_1.val, MOVE.PLAYER_2.val]).isRequired
};

export default PlayerInfo;
