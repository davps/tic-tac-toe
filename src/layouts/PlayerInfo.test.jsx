import React from 'react';
import renderer from 'react-test-renderer';
import { MOVE } from '../config';
import PlayerInfo from './PlayerInfo';

it('PlayerInfo_snapshot1', () => {
  // eslint-disable-next-line react/jsx-filename-extension
  const tree = renderer.create(<PlayerInfo player={MOVE.PLAYER_1.val} />);
  expect(tree).toMatchSnapshot();
});
it('PlayerInfo_snapshot2', () => {
  // eslint-disable-next-line react/jsx-filename-extension
  const tree = renderer.create(<PlayerInfo player={MOVE.PLAYER_2.val} />);
  expect(tree).toMatchSnapshot();
});
