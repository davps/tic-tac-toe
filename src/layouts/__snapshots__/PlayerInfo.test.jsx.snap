// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerInfo_snapshot1 1`] = `
<span>
   Jacob (X) 
</span>
`;

exports[`PlayerInfo_snapshot2 1`] = `
<span>
   David (O) 
</span>
`;
